src/core/qgsvectorlayerserverproperties.h

```diff
@@ -0,0 +1,150 @@
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSVECTORLAYERSERVERPROPERTIES_H
+#define QGSVECTORLAYERSERVERPROPERTIES_H
+
+#include "qgis_sip.h"
+#include "qgis_core.h"
+#include <QMap>
+#include <QString>
+#include <QMetaType>
+#include <QVariant>
+
+class QgsVectorLayer;
+
+class QDomNode;
+class QDomDocument;
+
+/**
+ * \ingroup core
+ * Manages QGIS Server properties for a vector layer
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsVectorLayerServerProperties
+{
+
+  public:
+
+    /**
+     * Setting to define QGIS Server WMS Dimension.
+     * \since QGIS 3.10
+     */
+    struct CORE_EXPORT WmsDimensionInfo
+    {
+
+      /**
+       * Selection behavior for QGIS Server WMS Dimension default display
+       * \since QGIS 3.0
+       */
+      enum DefaultDisplay
+      {
+        AllValues = 0, //!< Display all values of the dimension
+        MinValue = 1, //!< Add selection to current selection
+        MaxValue = 2, //!< Modify current selection to include only select features which match
+        ReferenceValue = 3, //!< Remove from current selection
+      };
+
+      /**
+       * Constructor for WmsDimensionInfo.
+       */
+      explicit WmsDimensionInfo( const QString &dimName,
+                                 const QString &dimFieldName,
+                                 const QString &dimEndFieldName = QString(),
+                                 const QString &dimUnits = QString(),
+                                 const QString &dimUnitSymbol = QString(),
+                                 const int &dimDefaultDisplayType = QgsVectorLayerServerProperties::WmsDimensionInfo::AllValues,
+                                 const QVariant &dimReferenceValue = QVariant() )
+        : name( dimName )
+        , fieldName( dimFieldName )
+        , endFieldName( dimEndFieldName )
+        , units( dimUnits )
+        , unitSymbol( dimUnitSymbol )
+        , defaultDisplayType( dimDefaultDisplayType )
+        , referenceValue( dimReferenceValue )
+      {}
+      QString name;
+      QString fieldName;
+      QString endFieldName;
+      QString units;
+      QString unitSymbol;
+      int defaultDisplayType;
+      QVariant referenceValue;
+    };
+
+    /**
+     * Constructor - Creates a Vector Layer QGIS Server Properties
+     *
+     * \param layer  The vector layer
+     */
+    QgsVectorLayerServerProperties( QgsVectorLayer *layer = nullptr );
+
+    /**
+     * Returns predefined/restricted WMS Dimension names
+     * \since QGIS 3.10
+     */
+    static QStringList predefinedWmsDimensionNames() { return { "time", "elevation" }; }
```

Comment:
I still think we should define an enum for predefined dimensions instead of playing with strings :).

Reply:
@pblottiere I'd like to do it in the struct, but it seems to be not that easy for me.

Reply:
OK, I'll try to take a look in the coming days.

Reply:
> I'll try to take a look in the coming days.

Sorry, I had my hands full these last days.

In the meantime, did you try to use an enum mechanism instead of strings in one of your branch? I could definitely take a look to give you a hand if you want.

Reply:
Done